plugin.rb

```diff
@@ -919,6 +919,10 @@ def check_enabled
     def index
       # guardian.ensure_can_use_data_explorer!
       queries = DataExplorer::Query.all
+      unless queries.any?{|q| q.created_by == Discourse::SYSTEM_USER_ID.to_s}
+        add_default_queries
```

Comment:
Take a DistributedMutex here and recheck the condition, otherwise you could end up with multiple copies of the queries created.

Reply:
Hi @riking! I'll now look into how DistributedMutex works. I didn't quite understand how multiple copies of the queries can be created, can you explain a bit? Thank you!